mllm/models/minicpm4/modeling_minicpm4.hpp

```
@@ -0,0 +1,393 @@
+
+    for (auto& block : blocks) { x = block(x, llm_embedding_sin, llm_embedding_cos, kv_cache)[0]; }
+
+    x = norm_(x);
+
+    return {x};
+  }
+};
+
+class MiniCPM4ForCausalLM : public ARGeneration, public nn::Module {
+ public:
+  explicit MiniCPM4ForCausalLM(const MiniCPM4Config& cfg) : cfg(cfg) {
```

Clang (14.0.6)

[error] constructor does not initialize these fields: lm_head_, tie_word_embeddings_, kv_cache_, longrope_scaling_factor_

(cppcoreguidelines-pro-type-member-init,-warnings-as-errors)